mongodump/prepare.go

```diff
@@ -341,7 +354,7 @@ func (dump *MongoDump) NewIntentFromOptions(dbName string, ci *db.CollectionInfo
 					Buffer: &bytes.Buffer{},
 				}
 			} else {
-				path := nameGz(dump.OutputOptions.Gzip, dump.outputPath(dbName, ci.Name+".metadata.json"))
+				path := nameGz(dump.OutputOptions.Gzip, dump.outputPath(dbName, ci.Name)+".metadata.json")
```

Comment:
Was this a pre-existing bug?

Reply:
Sort of? It didn't make any functional difference per se. Originally we were primarily joining the `colName` parameter with the `root` and `dbName`, so it didn't matter if you tossed in the filename extension or not.

It makes a difference now though when we count the letters in `colName` and when we hash it. The function's comment also says "sans file extension" for `colName`, so I had to make the adjustment here.

Reply:
Ok that makes sense. I took a look before commenting and saw that this was the odd-one-out in terms of uses of `nameGz`/`dump.outputPath` so I'm glad you fixed it. Just wanted to make sure it was something that needed to be changed and you confirmed that!